### PR TITLE
Fix an issue where destinations are compared against themselves

### PR DESCRIPTION
**Pull request by felipeogarcia**
The two calls to `Random.Next` may return the same index, so the comparison happens against the same destination. This behavior compromises the worst case protection this algorithm has because it might only use the worst destination, not offering opportunity of a better one. The fewer destinations we have, the more prominent this issue becomes.

### CODE REVIEW

**Comment by felipeogarcia** [comment]
@microsoft-github-policy-service agree

**Review by MihaZupan — APPROVED** [review]
Thanks

File: src/ReverseProxy/LoadBalancing/PowerOfTwoChoicesLoadBalancingPolicy.cs [hunk 1]
```diff
@@ -21,17 +21,28 @@ public PowerOfTwoChoicesLoadBalancingPolicy(IRandomFactory randomFactory)
...
-        var destinationCount = availableDestinations.Count;
         var random = _randomFactory.CreateRandomInstance();
-        var first = availableDestinations[random.Next(destinationCount)];
-        var second = availableDestinations[random.Next(destinationCount)];
+        var firstIndex = random.Next(destinationCount);
+        int secondIndex;
+        do
+        {
+            secondIndex = random.Next(destinationCount);
+        } while (firstIndex == secondIndex);
```
Comment (Tratcher) [inline src/ReverseProxy/LoadBalancing/PowerOfTwoChoicesLoadBalancingPolicy.cs:43]: This fix suffers from the same issue as the original, a high collision rate in small sets. I think it'd be better to do a deterministic shift in those cases.
```
int secondIndex = random.Next(destinationCount);
if (secondIndex == firstIndex)
{
    secondIndex = (irstIndex + 1) % destinationCount;
}
```

I'll send a PR.

Comment (Tratcher) [reply src/ReverseProxy/LoadBalancing/PowerOfTwoChoicesLoadBalancingPolicy.cs:43]: @MihaZupan convinced me that the +1 approach leads to non-trivial imbalance.

Even in the smallest case of two destinations where collisions will be common, the loop will only need to run a few times on average to break the tie. That's probably adequate.

**Pull request merged**